Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java

```diff
@@ -284,4 +342,24 @@ private void fetchKeyStore() {
     private void fetchTrustStore() {
         truststores = service.fetchTrustStores();
     }
+
+    public Certificate viewCertificate() {
+        if (selectedTSCertificate != null) {
+            oldAlias = selectedTSCertificate.getAlias();
+            RequestContext.getCurrentInstance().execute("PF('viewCertDlg').show();");
+        } else {
```

Comment:
viewCertificate and viewCertificateKS are basically the same method. You can move the common functionality into a private method and pass the dialog string as an argument to that method.